addons/repair/models/repair.py

```diff
@@ -191,6 +198,14 @@ def _onchange_company_id(self):
         else:
             self.location_id = False
 
+    def unlink(self):
+        for order in self:
+            if order.state not in ('draft', 'cancel'):
+                raise UserError(_('You can not delete a repair order once it has been confirmed. You must first cancel it.'))
```

Comment:
> once it has been confirmed. 

Is it allowed after canceling it? (Maybe we are mixing up here)

```suggestion
                raise UserError(_('In order to delete a Repair Order. You must first cancel it.'))
```